### DISCUSSION TITLE
Version 5.3.0

### DISCUSSION THREAD

**Discussion by jpivarski** [scikit-hep/uproot5#1143] [New releases]
## New features

* feat: use cramjam for lzma, lz4, and zstd, opt-in use of isal for zlib by @lgray in https://github.com/scikit-hep/uproot5/pull/1090
* feat: expose decompression_executor and interpretation_executor in uproot dask by @ioanaif in https://github.com/scikit-hep/uproot5/pull/1120

## Bug-fixes and performance

* fix: missing '_fh' and '_file' attributes after unpickling by @jpivarski in https://github.com/scikit-hep/uproot5/pull/1118
* fix: fix dask_write docs by @zbilodea in https://github.com/scikit-hep/uproot5/pull/1122
* fix: attempt to concatenate numpy and awkward arrays by @ioanaif in https://github.com/scikit-hep/uproot5/pull/1114
* fix: test suite creates files not in tmp_path by @ioanaif in https://github.com/scikit-hep/uproot5/pull/1123
* fix: allow colon in key names by @ioanaif in https://github.com/scikit-hep/uproot5/pull/1127
* fix: allow writing generic UHI-compatible histograms by @YSelfTool in https://github.com/scikit-hep/uproot5/pull/1128

## Other

* docs: add YSelfTool as a contributor for code by @allcontributors in https://github.com/scikit-hep/uproot5/pull/1131
* chore: update pre-commit hooks by @pre-commit-ci in https://github.com/scikit-hep/uproot5/pull/1110
* chore: update pre-commit hooks by @pre-commit-ci in https://github.com/scikit-hep/uproot5/pull/1119
* chore: update pre-commit hooks by @pre-commit-ci in https://github.com/scikit-hep/uproot5/pull/1133

## New Contributors
* @YSelfTool made their first contribution in https://github.com/scikit-hep/uproot5/pull/1128

**Full Changelog**: https://github.com/scikit-hep/uproot5/compare/v5.2.2...v5.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/scikit-hep/uproot5/releases/tag/v5.3.0'>Version 5.3.0</a>.</em>

**Comment by raymondEhlers** [comment]
Low priority question, but are there any further docs regarding #1090? From reading through the PR, It sounds like there are performance improvements, but some are opt-in only? Are there tradeoffs in doing so? To opt in, I just need to install the right package? (`isal`?). Thanks!

**Reply by jpivarski** [reply]
ISAL is the only performance improvement. The motivation for going to cramjam was portability. (In fact, the ISAL part is unrelated and technically should have been a different PR.)

It's in the documentation, though buried: https://uproot.readthedocs.io/en/latest/uproot.compression.ZLIB.html

> If ZLIB.library is "zlib" (default), Uproot uses zlib from the Python standard library.
>
> If ZLIB.library is "isal" (default), Uproot uses isal.isal_zlib.

(I'm fixing the docs failure: https://github.com/scikit-hep/uproot5/commit/d2464242b1545a81b30bfd7413e25102bb6650c8.)

For more detail on ISAL:

* [Intel technical note](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-isa-l-and-intel-integrated-performance-primitives-zlib-solutions.html)
* [Python library for ISAL](https://github.com/pycompression/python-isal?tab=readme-ov-file#readme)
* [StackOverflow issues](https://stackoverflow.com/q/77017255/1623645)

**Reply by raymondEhlers** [reply]
Okay, great - thanks for clarifying and for the links!
